### PR TITLE
Implement/refactor `getSchema` for the SQL prop

### PR DESCRIPTION
**Pull request by jverce**
# Changelog
* [Refactor getSchema in MySQL](https://github.com/PipedreamHQ/pipedream/commit/14295e076eced1a6586bb9226e37138ae6731198)
* [Refactor getSchema in PostgreSQL](https://github.com/PipedreamHQ/pipedream/commit/133a22320f1301ee81edfa68069c996c69d258eb)
* [Implement getSchema in Snowflake](https://github.com/PipedreamHQ/pipedream/commit/5b14a78646b2c95e75784cd45592a1190bfb1f13)

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/mysql/package.json
Comment (andrewjschuang) [inline components/mysql/package.json:1]: Should `@pipedream/platform` be updated to 2.0.0 here?

File: components/postgresql/package.json
Comment (andrewjschuang) [inline components/postgresql/package.json:1]: Should `@pipedream/platform` be updated to 2.0.0 here?

File: components/snowflake/package.json
Comment (andrewjschuang) [inline components/snowflake/package.json:1]: Should `@pipedream/platform` be updated to 2.0.0 here?

File: components/postgresql/postgresql.app.mjs [hunk 1]
```diff
@@ -228,8 +228,11 @@ export default {
         text,
       });
       return rows.reduce((acc, row) => {
-        acc[row.tableName] ??= {};
-        acc[row.tableName][row.columnName] = {
+        acc[row.tableName] ??= {
+          metadata: {},
```
Comment (andrewjschuang) [inline components/postgresql/postgresql.app.mjs:232]: Are we not able to get the `rowCount`?

**Review by andrewjschuang — COMMENTED** [review]
Thanks, added just a few comments

File: components/postgresql/postgresql.app.mjs (hunk 1 above)
Comment (jverce) [reply components/postgresql/postgresql.app.mjs:232]: Not for this DB, no unfortunately

**Review by andrewjschuang — APPROVED** [review]
lgtm!

**Pull request merged**